CRM/Contribute/BAO/Contribution.php

```diff
@@ -4109,6 +4109,18 @@ public static function checkTaxAmount($params, $isLineItem = FALSE) {
 
     // Update contribution.
     if (!empty($params['id'])) {
+      // CRM-19126, civicrm_line_item.tax_amount incorrectly set when using online payment processor.
+      // It looks like this method is meant to be called in multiple contexts: new & update
+      // When creating, would expect total_amount to be set?  Prior to 4.7, when creating online membership
+      // via contribution page, call scenario was different.
+      // This would not never get called, end result, taxes were correct.
+      // Since 4.7 it does.  Not sure this fix is ideal, but it does do the trick.
+      // Conceptually, if we're "updating", and total_amount is unknown.  We're dealing with an incomplete
+      // view, why are we nulling out all line item taxes, or messing with any other tax amounts?
+      if (!isset($params['total_amount'])) {
+          return $params;
```

Comment:
I'm just wondering if its worth while moving this down a bit to under $prevContrivutionValue and saying if not set in the params set total_amount to be the same as the previous transaction like what happens with financial_type_id in line 4130 - 1432